test/util.ts

```diff
@@ -68,3 +73,62 @@ export function hex2arrayBuffer(data: string): ArrayBuffer {
 
     return ret.buffer
 }
+
+export const verifyPackedAttestation = async (keys: CryptoKeyPair, webAuthnResponse: PublicKeyCredential) => {
+    if (process.versions.node < '16') {
+        console.error('not support import jwk lower than nodejs v16')
+        console.error('ignoring the test')
+        return true
+    }
+    const attestationBuffer = Buffer.from(webAuthnResponse.response.attestationObject)
+    const attestationStruct = decode(attestationBuffer.buffer)
+
+    const authDataStruct = parseAuthData(attestationStruct.authData)
+
+    const clientDataJSONBuffer = Buffer.from(webAuthnResponse.response.clientDataJSON)
+    const clientDataHash = Buffer.from(await sha256(clientDataJSONBuffer))
+    const originalDataBuffer = Buffer.concat([attestationStruct.authData, clientDataHash])
+
+    const signatureBuffer = attestationStruct.attStmt.sig
+    let signatureIsValid = false
+
+    if (!authDataStruct.COSEPublicKey) {
+        throw new TypeError('COSE Public Key not found')
+    }
+
+    let publicKeyCose = decode(authDataStruct.COSEPublicKey.buffer) as EcCosePublicKey
+    const alg = Alg[publicKeyCose[CoseKey.alg]]
+    if (publicKeyCose[CoseKey.kty] === Kty.EC) {
+        const x = publicKeyCose[CoseKey.x]
+        const y = publicKeyCose[CoseKey.y]
+        const crv = Crv[publicKeyCose[CoseKey.crv]]
+        const kty = Kty[publicKeyCose[CoseKey.kty]]
+
+        // start verify
+        const publicKey = NodeCrypto.createPublicKey({
+            format: 'jwk',
+            key: {
+                x,
+                y,
+                crv,
+                kty,
+                alg,
+            },
+        })
+
+        // todo: refactor this to node implementation
```

Comment:
I haven't learned how to use `verify` in nodejs, the nodejs API is too complex and unfriendly docs

Reply:
this is webcrypto api, maybe MDN doc helps

Reply:
Yes, but i want to use this api https://nodejs.org/dist/latest-v16.x/docs/api/crypto.html#crypto_crypto_verify_algorithm_data_key_signature_callback